app/models/map.rb

```diff
@@ -96,6 +96,10 @@ def self.featured
        .order('image_count DESC')
   end
 
+  def self.location(loc)
+    Map.where(location: loc)
```

Comment:
At last, you're fetching maps using this query so directly use it in the controller instead of calling helper and then model method.

Or, if you want, just create a method. Also, use Like in your query https://melvinchng.github.io/rails/SearchFeature.html#45-wildcards-for-like-operator as current query looks for full name

Reply:
I have made the changes.